13/src/academy/pocu/comp2500samples/w13/stack/Stack.java

```diff
@@ -0,0 +1,29 @@
+package academy.pocu.comp2500samples.w13.stack;
+
+import java.util.ArrayList;
+
+public final class Stack<E> extends ArrayList<E> {
+    public void push(E element) {
+        this.add(element);

```

Comment:
i think the intension was this example was overriding arraylist's add and remove function so taht it works as stack..

once u do that u cannot use Stack in place for ArrayList